src/common/Form/index.jsx

```diff
@@ -1,63 +1,105 @@
 import styles from './index.module.css'
 import { Button } from '..'
 import { sendMessageAction } from '../../actions'
+import { isEmailValid } from './utils'
+
+const initialUserState = {
+  email: '',
+  reason: '',
+  nick: '',
+  content: '',
+}
 
 function Form() {
   const dispatch = useDispatch()
   const navigate = useNavigate()
 
-  const [email, setEmail ] = useState('')
-  const [reason, setReason ] = useState('')
-  const [name, setName ] = useState('')
-  const [content, setContent ] = useState('')
+  const [user, setUser] = useState(initialUserState)
   const [checked, setChecked] = useState(true)
 
-  const handleInput = (e, fun)  => {
-    fun(e.target.value)
+  const handleChange = (e) => {
+    const name = e.target.name
+    const value = e.target.value
+    setUser({ ...user, [name]: value })
   }
 
   const handleSubmit = (e) => {
-    const message = {
+    e.preventDefault()
+    if (!isEmailValid(user.email)) {
+      return null
+    }
+    if (user.nick && user.reason && user.email && user.content) {
+      const message = {
         person: {
-          email: email,
-          name: name
+          email: user.email,
+          name: user.nick,
         },
-        message: content,
-        department: reason,
+        message: user.content,
+        department: user.reason,
         newsletter: checked,
+      }
+      setUser(initialUserState)
+      dispatch(sendMessageAction(message))
+      navigate('/submitmessage')
     }
-    e.preventDefault()
-    dispatch(sendMessageAction(message))
-    navigate('/submitmessage')
-  }
-
-  const handleCheckbox = () => {
-    setChecked(!checked)
-  }
-
-  const handleCheckboxChange = () => {
-    return ''
   }
 
   return (
-    <form className={styles.root} id='contact'>
-        <input className={styles.input} type="text" placeholder="Email"  onChange={(e) => handleInput(e,setEmail)} value={email}/>
-        <select className={styles.select} type="text" placeholder="Reason for contact" onChange={(e) => handleInput(e, setReason)} value={reason}>
-          <option value='none'>Choose an option</option>
-          <option value='volunteer'>Volunteer with us!</option>
-          <option value='perform'>Perform at our Pride 2023</option>
-        </select>
-        <input className={styles.input} type="text" placeholder="Name" onChange={(e) => handleInput(e, setName)} value={name}/>
-        <textarea rows='14' cols='10' className={styles.areaInput} placeholder="Your message"  onChange={(e) => handleInput(e, setContent)} defaultValue={content}></textarea>
-        <div className={styles.buttonParent}>
-            <Button text="Submit" variant="dark" handleClick={handleSubmit}/>
-        </div>
+    <form className={styles.root} id='contact' onSubmit={handleSubmit}>
+      <input
+        className={styles.input}
+        type='text'
+        placeholder='Email'
+        name='email'
+        onChange={handleChange}
+        value={user.email}
+        required
+      />
```

Comment:
When the email is in incorrect format, the red text should appear underneath the imput - 'please enter a valid email'